includes/multi-currency/Compatibility/WooCommerceProductAddOns.php

```diff
@@ -248,23 +309,33 @@ public function order_line_item_meta( array $meta_data, array $addon, \WC_Order_
 				$addon_price = $this->multi_currency->get_price( $addon['price'], 'product' );
 			}
 			if ( class_exists( '\WC_Product_Addons_Helper' ) ) {
-				$price          = html_entity_decode(
+				// phpcs:ignore
+				/**
+				 * @psalm-suppress UndefinedClass
+				 */
+				$price = html_entity_decode(
 					wp_strip_all_tags( wc_price( \WC_Product_Addons_Helper::get_product_addon_price_for_display( $addon_price, $values['data'] ) ) ),
 					ENT_QUOTES,
 					get_bloginfo( 'charset' )
 				);
-				$addon['name'] .= ' (' . $price . ')';
 			}
-		}
 
-		if ( 'custom_price' === $addon['field_type'] ) {
-			$addon['value'] = $addon['price'];
+			if ( 'flat_fee' === $addon['price_type'] && $addon['price'] && $add_price_to_value ) {
+				/* translators: %1$s flat fee addon price in order */
+				$value .= sprintf( _x( ' (+ %1$s)', 'flat fee addon price in order', 'woocommerce-payments' ), $price );
```

Comment:
We may want to consider handling things more gracefully when `$price` is not defined. Which can happen if `class_exists( '\WC_Product_Addons_Helper' )` resolves to `false`.

Reply:
I thought about this, but couldn't find a case where the `WC_Product_Addons_Helper` wouldn't be defined -- do you have a specific case in mind? 

Right now, if this class is not defined, the the add-on value will be the same as the one set by Product Add-Ons, i.e. not converted by the selected currency. Would you prefer to have a different fallback here?

Reply:
Although it's unlikely, this is one scenario where this may happen:
* We remove `WC_Product_Addons_Helper` in a future version of `woocommerce-product-addons`.
* The merchant installs the future version of `woocommerce-product-addons`.
* The merchant installs this version of WooPayments.

Although this is possible, it's unlikely, so we can proceed without making modifications.